cs/di-extensions.texy

```diff
@@ -5,73 +5,115 @@ DI: Rozšíření pro kontejner
 
 /---code neon
 extensions:
-	blog: MyBlogExtension
+	blog: BlogExtension
 \---
 
-Každé rozšíření `Compiler`u musí dědit od [Nette\DI\CompilerExtension|api:] a může implementovat tři různé metody, které jsou postupně volány, během sestavování `Container`u.
+Každé rozšíření `Compiler`u musí dědit od [Nette\DI\CompilerExtension|api:] a může implementovat tři různé metody, které jsou postupně volány, během sestavování `Container`u:
 
+1. loadConfiguration()
+2. beforeCompile()
+3. afterCompile()
 
-CompilerExtension::loadConfiguration() .{toc: loadConfiguration()}
-==================================================================
 
-Metoda je nad všemi rozšířeními volána jako první a je určena k načtení dodatečných konfiguračních souborů, vytvoření dalších služeb pomocí [Nette\DI\ContainerBuilder|api:] a hlavně zpracování konfigurace aplikace.
+loadConfiguration() .[method]
+=============================
 
-V configu je možné uvést sekci, která se jmenuje stejně jako naše rozšíření. Pokud tedy budeme uvažovat předchozí příklad, mohlo by v konfigu přibýt například toto
+Tato metoda se volá jako první a je určena k přidání služeb souvisejících s daným rozšířením.
+
+Na registrování služeb použijeme [Nette\DI\ContainerBuilder|api:].
+
+/--php
+class BlogExtension extends Nette\DI\CompilerExtension
+{
+	public function loadConfiguration()
+	{
+		$builder->addDefinition($this->prefix('articles'))
+			->setFactory('App\Model\HomepageArticles', ['@connection'])
+			->addSetup('setLogger', ['@logger']);
+	}
+}
+\--
+
+.[tip]
+Konvence je, prefixovat služby v rozšíření jeho názvem, aby nevznikaly konflikty. Pomůže nám s tím metoda `prefix()`.
```

Comment:
Konvence je prefixovat služby přidané rozšířením jeho názvem, aby nevznikaly jmenné konflikty. To dělá metoda `prefix()`, takže pokud se rozšíření jmenuje `blog`, služba ponese název `blog.articles`.